tidoop-mr-lib/README.md

```diff
@@ -1,6 +1,240 @@
-#Tidoop - MapReduce job library
+#<a name="top"></a>Tidoop - MapReduce job library
 
-##Contact
+* [What is tidoop-mr-lib](#whatis)
+* [Building](#mainbuilding)
+    * [Prerequisites](#prerequisites)
+    * [Building tidoop-mr-lib and its dependencies](#building)
+    * [Unit tests](#unittests)
+* [Installation](#install)
+* [Usage](#usage)
+    * [Running in the command line](#commandline)
+    * [Oozie-based running](#oozie)
+    * [Using tidoop-mr-lib-api](#mrlibapi)
+* [Library details](#details)
+    * [About input and output](#inputoutput)
+    * [Job categories](#categories)
+        * [Transformation and computation jobs](#transformationcomputation)
+        * [Generic and NGSI specific jobs](#genericngsi)
+        * [Flat and key-value jobs](#flatkeyvalue)
+    * [Chaining of jobs](#chaining)
+* [Jobs reference (in alphabetical order)](#jobs)
+    * [Filter](#filter)
+* [Contact](#contact)
+
+##<a name="whatis"></a>What is tidoop-mr-lib
+
+tidoop-mr-lib is a library of already developed-ready to use [MapReduce](http://research.google.com/archive/mapreduce.html) jobs for [Hadoop](https://hadoop.apache.org), allowing data scientists and engineers in genaral for an easy and quick approach to Big Data analysis.
+
+Available jobs can be used either to transform an input dataset, either to compute a value from it. Usually, these transformations or computations are suited for any general dataset, nevertheless there are NGSI specific MapReduce jobs simplifying a lot the process of analyzing NGSI-like context data. Have a look on this and many other ways of classifying the jobs in the [library details](#details) section.
+
+Finally, it is worth to mention MapReduce jobs available within this library can be combined in a chain of executions, linking the outout of a job to another one's input. Job chaining can be done from the command line, manually running jobs sequentially, using [tidoop-mr-lib-api](../tidoop-mr-lib-api) or using [Apache Oozie](http://oozie.apache.org/). See the [chaining of jobs](#chaining) section for further details.
+
+[Top](#top)
+
+##<a name="mainbuilding"></a>Building
+###<a name="prerequisites"></a>Prerequisites
+
+Maven (and thus Java SDK, since Maven is a Java tool) is needed in order to build tidoop-hadoop-ext.
+
+In order to install Java SDK (not JRE), just type (CentOS machines):
```

Comment:
Instead of constraining to a specific version of CentOS (even, CentOS is not mandatory, it could be any Linux distribution), I'll made more generic the commands were a specific version (e.g. Java SDK) appears.